<a href="https://colab.research.google.com/github/prasamshatripathi/Basic-Form/blob/main/google_collaboration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

system specification

In [ ]:
!cat /proc/cpuinfo


processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 79
model name	: Intel(R) Xeon(R) CPU @ 2.20GHz
stepping	: 0
microcode	: 0xffffffff
cpu MHz		: 2199.998
cache size	: 56320 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx smap xsaveopt arat md_clear arch_capabilities
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs taa mmio_stale_data retbleed bhi
bogomips	: 4399.99
clflush size	: 64
cache_alignment	: 64
ad

In [ ]:
# !cat /proc/meminfo

In [ ]:
!cat /proc/meminfo

MemTotal:       13290460 kB
MemFree:         7225892 kB
MemAvailable:   12106428 kB
Buffers:          417600 kB
Cached:          4593776 kB
SwapCached:            0 kB
Active:           711524 kB
Inactive:        5001220 kB
Active(anon):       1380 kB
Inactive(anon):   701728 kB
Active(file):     710144 kB
Inactive(file):  4299492 kB
Unevictable:           8 kB
Mlocked:               8 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:               476 kB
Writeback:             0 kB
AnonPages:        701336 kB
Mapped:           343120 kB
Shmem:              1732 kB
KReclaimable:     196304 kB
Slab:             240956 kB
SReclaimable:     196304 kB
SUnreclaim:        44652 kB
KernelStack:        5908 kB
PageTables:        11436 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     6645228 kB
Committed_AS:    3022956 kB
VmallocTotal:   34359738367 kB
VmallocUsed:       12280 kB
VmallocChunk:    

installing library

In [ ]:
!pip install pandas

This line imports the pandas library and assigns it the alias "pd", for convenient and concise usage within your code.

In [ ]:
import pandas as pd

This line of code reads the data from the specified CSV file into a pandas DataFrame and stores it in a variable named df, allowing you to further process and analyze the data.

In [11]:
df=pd.read_csv('/content/sample_data/california_housing_test.csv')

In [12]:
df.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0    -122.05     37.37                27.0       3885.0           661.0   
1    -118.30     34.26                43.0       1510.0           310.0   
2    -117.81     33.78                27.0       3589.0           507.0   
3    -118.36     33.82                28.0         67.0            15.0   
4    -119.67     36.33                19.0       1241.0           244.0   

   population  households  median_income  median_house_value  
0      1537.0       606.0         6.6085            344700.0  
1       809.0       277.0         3.5990            176500.0  
2      1484.0       495.0         5.7934            270500.0  
3        49.0        11.0         6.1359            330000.0  
4       850.0       237.0         2.9375             81700.0

In [13]:
print("machine learning")

machine learning


In [14]:
print(8)

8


if we are not sure about datatype we can use keyword'tyype'.

In [15]:
type(8)

int

concatenation method

In [16]:
print ("machine learning" + "project")

machine learningproject


In [17]:
print ("machine learning" + " project")

machine learning project


In [18]:
type("english")

str

space is requare after double inverted comma in project to sepearte learning and project


# **constants and variables**

if one variable used 'superhero'

In [19]:
superhero="ironman"
print(superhero)

ironman


if multiple variables were used

In [20]:
hero1,hero2,hero3="ironman","thor","captain"
print(hero1)
print(hero2)
print(hero3)

ironman
thor
captain


# ***Function:input function***

In [21]:
number1=input("enter the first number")
number2=input("enter the second number")
sum=number1+number2
print(sum)

enter the first number34
enter the second number56
3456


we didn't get 90, instead we got 3456 because they were concatenated because system recognize our input as string so we need to mention datatype before input function to get integer sum value

In [22]:
number1=int(input("enter the first number"))
number2=int (input("enter the second number"))
sum=number1+number2
print(sum)

enter the first number56
enter the second number34
90


# *changing datatypes*

In [23]:
num=5
print(float(num))

5.0


# **Types of object in python**

### 1.immutable object
### 2.mutable objects

# Immutable objects cannot be changed in place, while mutable objects can be modified directly.
immutable=int,float,string,bool,tuple
mutable=list,set,dictionary

### **List**

In [24]:
#list should be included in the square brackets
my_list=[1,2,3,4,5]
print(my_list)
type(my_list)

[1, 2, 3, 4, 5]


list

In [25]:
#list can have multiple datatypes
my_list=[2,3,1.8,'english',True]
print(my_list)
type(my_list)

[2, 3, 1.8, 'english', True]


list

In [26]:
#lists are mutable--> Changeable
#adding element to list
my_list=[2,3,1.8,'english',True]
my_list.append(6)
print(my_list)


[2, 3, 1.8, 'english', True, 6]


# .append is used to add new element in list

In [27]:
#print elements of a list using their index
print(my_list[0])
print(my_list[2])

2
1.8


In [32]:
#lists allow duplicate values
list_1=[1,2,3,4,5,2,3,4]
print(list_1)
#to find length or size of list use len function
print(len(list_1))

[1, 2, 3, 4, 5, 2, 3, 4]
8


In [33]:
#initiating a empty list
list2=[]
print(list2)
#we can add element using .append to listname

[]


In [34]:
list2.append(5)
print(list2)

[5]


In [36]:
#delete an item in a list
list_1=[1,2,3,4,5,2,3,4]
print (list_1)
del list_1[2]
print(list_1)

[1, 2, 3, 4, 5, 2, 3, 4]
[1, 2, 4, 5, 2, 3, 4]


In [37]:
#join two lists
list_3=[1,2,3,4]
list_4=[5,6,7,8]
list_5=list_3+list_4
print(list_5)


[1, 2, 3, 4, 5, 6, 7, 8]


# **Tuples**

Tuples are ordered, immutable collections of elements, similar to lists but unchangeable after creation. Created using parentheses (), they can hold diverse data types. Useful for representing fixed data, returning multiple values, and dictionary keys due to hashability.

In [38]:
tuple_1=(1,2,3,4)
print(tuple_1)
type(tuple_1)


(1, 2, 3, 4)


tuple

In [39]:
#tuple allows multiple datatypes
tuple2=(1,2,3.5,'machine learning',False)
print(tuple2)

(1, 2, 3.5, 'machine learning', False)


In [40]:
#converting a list into tuple
my_list=[3,4,5,6]
print(my_list)
my_tuple=tuple(my_list)
print(my_tuple)


[3, 4, 5, 6]
(3, 4, 5, 6)


In [41]:
print (my_tuple[0])
print (my_tuple[2])

3
5


Tuples are immutable

In [42]:
my_tuple.append(6)

AttributeError: 'tuple' object has no attribute 'append'

error due to immutability feature of tuple so cannot add new item

In [43]:
my_tuple=(1,2,3,8,7,6,8)
print(len(my_tuple))

7


#**Set**

Sets are unordered collections of unique elements. Created using curly braces {} or the set() function, they automatically remove duplicates. Useful for removing duplicates, set operations, and membership testing.

In [44]:
set={1,2,3,4,5}
print(set)
type(set)

{1, 2, 3, 4, 5}


set

In [45]:
#indexing is not supported in sets
print(set[0])

TypeError: 'set' object is not subscriptable

In [51]:
from re import X
#convert a list to a set
list_5= [4,5,6,7,8]
print(list_5)
x=set(list_5)
print(x)


[4, 5, 6, 7, 8]


TypeError: 'set' object is not callable

In [52]:
#set doesnot allow duplicate values
set_1={1,2,3,4,3,4,3,5,4,5,6}
print(set_1)

{1, 2, 3, 4, 5, 6}
